### PR TITLE
feat: implement card component

### PR DESCRIPTION
**Pull request by ledutu2**
## Check List
- [x] Your code builds clean without any errors or warnings.
- [x] Check coding convention.
- [x] Updated Storybook components and stories to reflect any changes made to the UI.
- [x] Test covers all edge cases and coverage is greater than or equal to 80%.

## Proof of Completeness
Screenshot changes introduced" width="250" />
OR
Video link

### CODE REVIEW

File: example/src/stories/Card.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import React from 'react'
+import type {ComponentMeta, ComponentStory} from '@storybook/react'
+
```
Comment (huydosgtech) [inline example/src/stories/Card.stories.tsx:3]: remove empty space

**Review by huydosgtech — COMMENTED** [review]
Got some stylelint code. Please fix it

**Comment by TranQuangPhi** [comment]
Please write some test code for it

**Comment by hangnguyensaigontech** [comment]
put Card.tsx into folder same as other PRs

File: src/components/Card.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import styled from 'styled-components/native'
+import {metrics} from '../helpers/metrics'
+import type {ITheme} from '../theme'
+import type {ViewProps} from 'react-native'
+
+export type CardProps = {
+  onPress?: () => void
+  children?: React.ReactNode
+  bgColor?: string
+  activeOpacity?: number
```
Comment (hangnguyensaigontech) [inline src/components/Card.tsx:11]: why define activeOpacity here but don't use in component?

File: src/components/Card.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  style?: ViewProps
+}
+
+const Card: React.FC<CardProps> = ({onPress, children, bgColor, style}) => (
+  <View onPress={onPress} activeOpacity={onPress ? 0.8 : 1} bgColor={bgColor} style={style}>
```
Comment (hangnguyensaigontech) [inline src/components/Card.tsx:16]: define constant for number

**Review by baonguyenhsts — CHANGES_REQUESTED** [review]

File: src/components/Card/Card.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    style={style}
+    testID={testId}>
+    {children}
+  </View>
+)
+
+const View = styled.TouchableOpacity`
+  ${(props: {theme: ITheme; bgColor: string | undefined}) => ({
+    padding: metrics.small,
+    borderRadius: metrics.borderRadius,
```
[the remark this answers is not in the document]
Comment (ledutu2) [reply src/components/Card/Card.tsx:29]: I think we can custom style if we want. @baonguyenhsts

[the remark this answers is not in the document]
Comment (baonguyenhsts) [reply src/components/Card/Card.tsx:29]: I mean you let the user pass the bgColor so the padding or borderRadius should be the same.
If the user doesn't want to update it, the default value will be metrics.small or anything.

File: src/theme/base/colors.ts [hunk 5]
```diff
@@ -2,6 +2,8 @@ const colors = {
   mainBackground: '#F0F2F3',
   cardPrimaryBackground: '#0A906E',
   textColor: '#0B0B0B',
+  white: '#ffffff',
+  cardBackground: '#ffffff',
```
Comment (ThinhKimVo) [inline src/theme/base/colors.ts:6]: why we need to separate the name, while they are the same color.

File: src/components/Card/Card.tsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+import {activeOpacity, metrics} from '../../helpers/metrics'
+import type {ITheme} from '../../theme'
+import type {ViewProps} from 'react-native'
+
+export type CardProps = {
+  onPress?: () => void
+  children?: React.ReactNode
+  bgColor?: string
+  style?: ViewProps
+  testId?: string
```
Comment (anhle10051996) [inline src/components/Card/Card.tsx:12]: do we need to pass testId props ?

File: src/components/Card/Card.tsx [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+    onPress={onPress}
+    activeOpacity={onPress ? activeOpacity.low : activeOpacity.none}
+    bgColor={bgColor}
+    style={style}
+    testID={testId}>
+    {children}
+  </View>
+)
+
+const View = styled.TouchableOpacity`
```
Comment (anhle10051996) [inline src/components/Card/Card.tsx:26]: use styled(TouchableOpacity)(props)=>({})

File: src/components/Card/Card.tsx [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+  bgColor?: string
+  style?: ViewProps
+  testId?: string
+}
+
+const Card: React.FC<CardProps> = ({onPress, children, bgColor, style, testId}) => (
+  <View
+    onPress={onPress}
+    activeOpacity={onPress ? activeOpacity.low : activeOpacity.none}
+    bgColor={bgColor}
```
Comment (anhle10051996) [inline src/components/Card/Card.tsx:19]: why you need to define separate bgColor props ?

File: src/components/Card/Card.tsx [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+export type CardProps = {
+  onPress?: () => void
+  children?: React.ReactNode
+  bgColor?: string
+  style?: ViewProps
+  testId?: string
+}
+
+const Card: React.FC<CardProps> = ({onPress, children, bgColor, style, testId}) => (
+  <View
```
Comment (anhle10051996) [inline src/components/Card/Card.tsx:16]: i think you should name it  CardContainer or CardWrapper ... would be better instead of view which actually is a TouchableOpacity

File: src/components/Card/Card.tsx (hunk 6 above)
Comment (huydosgtech) [reply src/components/Card/Card.tsx:12]: i think we need pass testId  props because maybe we have to use many Card in one Screen

File: src/components/Card/Card.tsx (hunk 9 above)
Comment (huydosgtech) [reply src/components/Card/Card.tsx:16]: ok I updated it.

File: src/components/Card/Card.tsx (hunk 8 above)
Comment (huydosgtech) [reply src/components/Card/Card.tsx:19]: ok. I removed it

File: src/components/Card/Card.tsx (hunk 7 above)
Comment (huydosgtech) [reply src/components/Card/Card.tsx:26]: ok I updated it

File: src/components/Card/Card.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (huydosgtech) [reply src/components/Card/Card.tsx:29]: Should user add that styles in **style** default? So we dont need scare about what user want to input.
ex:  style: {borderWidth: 1, borderColor: 'red', margin: 10},

File: src/theme/base/colors.ts (hunk 5 above)
Comment (huydosgtech) [reply src/theme/base/colors.ts:6]: Yes. I removed it

File: src/components/Card.tsx (hunk 3 above)
Comment (huydosgtech) [reply src/components/Card.tsx]: yes, updated. Please help me check again

File: src/components/Card.tsx (hunk 2 above)
Comment (huydosgtech) [reply src/components/Card.tsx]: Updated it

File: example/src/stories/Card.stories.tsx [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+import React from 'react'
+import {Text} from 'react-native'
+import {Card} from 'rn-base-component'
+
+export default {
+  title: 'components/Card',
+  component: Card,
+} as ComponentMeta<typeof Card>
+
+export const Basic: ComponentStory<typeof Card> = args => (
```
Comment (honghoangsts) [inline example/src/stories/Card.stories.tsx:11]: don't use `args` we agree to use `rest`

**Review by baonguyenhsts — APPROVED** [review]

**Review by honghoangsts — CHANGES_REQUESTED** [review]

**Review by honghoangsts — APPROVED** [review]
